### ISSUE TITLE
ODIOU loss and its training time

### ISSUE THREAD

**Issue by colian** [Vegeta2020/SE-SSD#16]
Hello, thanks for your work!
I cannot install your codebase for some unknown reasons(docker environment), so I just copy your code. As for ODIOU loss, 
`self.odiou_3d_loss(boxes1, boxes2, weights, batch_size=4)`
It need to run this only one line for 7 minutes (KITTI, SECOND model)! 

Besides, the result of the first iter is also Irregular (train from sketch):
`odiou_loss[odiou_loss>0].shape=1162, weights[weights>0].shape==512, odiou_loss[weights>0].sum()==0). `

So, is the calculation time of ODIOU loss normal? And is the iou_loss output right(cannot be zero)?

**Comment by Vegeta2020** [comment]
It may be caused by too much calculation, in our codebase, the pre-trained model would lead to very limited boxes for calculation. I guess the iou_loss is hard to be zero.

**Comment by colian** [comment]
> It may be caused by too much calculation, in our codebase, the pre-trained model would lead to very limited boxes for calculation. I guess the iou_loss is hard to be zero.

Thank you for your reply!
I have another question, it points that orientation (\Delta r) is an important term, but i could not find any ablation study on this term. Since basically the IOU term already could  optimize the orientation, can you share the performance without extra optimization of (\Delta r)?

**Comment by Vegeta2020** [comment]
Yes, you may refer the results in our supplementary materials, where we conducted some ablation study on it.

**Comment by colian** [comment]
Thanks, how about \gamma is zero? or you didn't consider this sutuation?

**Comment by Vegeta2020** [comment]
Sure. If \gamma becomes zero, the result would become worse.

**Comment by colian** [comment]
Fine, thank you for your reply.
